packages/odyssey-react-mui/src/ThemeContrastProvider.tsx

```diff
@@ -0,0 +1,156 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, {
+  createContext,
+  useContext,
+  useRef,
+  useLayoutEffect,
+  useState,
+  useMemo,
+  useCallback,
+  ReactNode,
+} from "react";
+import * as Tokens from "@okta/odyssey-design-tokens";
+
+export type ContrastMode = "lowContrast" | "highContrast";
+
+export type ContrastModeContextType = {
+  contrastMode: ContrastMode;
+  parentBackgroundColor: string;
+};
+
+const ThemeContrastContext = createContext<ContrastModeContextType>({
+  contrastMode: "lowContrast",
+  parentBackgroundColor: "",
+});
+
+export const useThemeContrastContext = () => useContext(ThemeContrastContext);
+
+const hexToRgb = (hex: string): string => {
+  const bigint = parseInt(hex.slice(1), 16);
+  const r = (bigint >> 16) & 255;
+  const g = (bigint >> 8) & 255;
+  const b = bigint & 255;
+  return `rgb(${r}, ${g}, ${b})`;
+};
+
+export const hueNeutral50Rgb = hexToRgb(Tokens.HueNeutral50);
+
+const isTransparentColor = (color: string): boolean =>
+  color === "rgba(0, 0, 0, 0)" || color === "transparent";
+
+const normalizeRgbaToRgb = (rgba: string): string =>
+  rgba.replace(/rgba\((\d+), (\d+), (\d+), \d+\)/, "rgb($1, $2, $3)");
+
+const getElementComputedBackgroundColor = (element: HTMLElement): string =>
+  window.getComputedStyle(element).backgroundColor;
+
+const normalizeBackgroundColor = (bgColor: string): string => {
+  if (/rgba\((\d+), (\d+), (\d+), \d+\)/.test(bgColor)) {
+    const normalizedColor = normalizeRgbaToRgb(bgColor);
+    return normalizedColor === hueNeutral50Rgb
+      ? Tokens.HueNeutral50
+      : normalizedColor;
+  }
+  return bgColor === hueNeutral50Rgb ? Tokens.HueNeutral50 : bgColor;
+};
+
+/**
+ * Determines the effective background color of an element.
+ *
+ * @param element - The HTML element to check.
+ * @returns The effective background color. Returns "#ffffff" if no non-transparent background is found.
+ *
+ * Note:
+ * - Low contrast mode is used for white background (#ffffff or HueNeutralWhite).
+ * - High contrast mode is used for gray background (#f4f4f4 or HueNeutral50).
+ */
+export const getBackgroundColor = (element: HTMLElement | null): string => {
+  while (element) {
+    const bgColor = getElementComputedBackgroundColor(element);
+    if (!isTransparentColor(bgColor)) {
+      return normalizeBackgroundColor(bgColor);
+    }
+    element = element.parentElement;
+  }
+  return "#ffffff"; // Default to white/low contrast if no background color is found
+};
+
+type ThemeContrastProviderProps = {
+  children: ReactNode;
+  contrastMode?: ContrastMode;
+};
+
+export const ThemeContrastProvider: React.FC<ThemeContrastProviderProps> = ({
+  children,
+  contrastMode: explicitContrastMode,
+}) => {
```

Comment:
We don't need to use `React.FC` here, and we don't in other places, so I'd prefer we remove it.

```suggestion
export const ThemeContrastProvider = ({
  children,
  contrastMode: explicitContrastMode,
}: ThemeContrastProviderProps) => {
```